server/backend/database/testcases/testcases.go

```diff
@@ -93,6 +93,41 @@ func RunFindDocInfoTest(
 	})
 }
 
+// RunFindDocInfosByKeysTest runs the FindDocInfosByKeys test for the given db.
+func RunFindDocInfosByKeysTest(
+	t *testing.T,
+	db database.Database,
+	projectID types.ID,
+) {
+	t.Run("find docInfos by keys test", func(t *testing.T) {
+		ctx := context.Background()
+		clientInfo, err := db.ActivateClient(ctx, projectID, t.Name())
+		assert.NoError(t, err)
+
+		// 01. Create documents
+		docKeys := []key.Key{
+			"test", "test$3", "test123", "test$0",
+			"search$test", "abcde", "test abc",
+		}
+		for _, docKey := range docKeys {
+			_, err := db.FindDocInfoByKeyAndOwner(ctx, clientInfo.RefKey(), docKey, true)
+			assert.NoError(t, err)
+		}
+
+		// 02. Find documents
+		infos, err := db.FindDocInfosByKeys(ctx, projectID, docKeys)
+		assert.NoError(t, err)
+
+		actualKeys := make([]key.Key, len(infos))
+		for i, info := range infos {
+			actualKeys[i] = info.Key
+		}
+
+		assert.ElementsMatch(t, docKeys, actualKeys)
+		assert.Len(t, infos, len(docKeys))
+	})
```

Comment:
It seems that MemDB and MongoDB implementations behave differently for non-existent keys. How about adding the corresponding test code?

Reply:
Oh, I see. Thanks for the review. In the mongo implementation of `FindDocInfoByKey()`, it returns an `ErrDocumentNotFound` error if there is no DocInfo for the given key. This suggests that the business logic requires a DocInfo to always be present for the requested key.

Should this business rule be followed for `FindDocInfosByKeys()` as well? That is, if any key does not have a corresponding DocInfo, should it return an `ErrDocumentNotFound` error?

If the above approach is appropriate, I will write test code with this scenario.

Reply:
I think there are two approaches for handling missing documents in GetDocuments API and each have their own advantages and disadvantages.

A. Returning the remaining documents even if some are missing
- Pros: More flexible and user-friendly as it returns information for available documents even if some are missing
- Cons: Behaves differently from the `GetDocument API`

B. Returning an error for missing documents
- Pros: Keeps consistency with `GetDocument API` by returning an error for non-existent keys
- Cons: Lacks flexibility as the entire request fails if even one document is missing

Personally, I prefer option A approach.

Reply:
Listing the pros and cons has helped me consider the trade-offs.

It seems like a choice between maintaining consistency and ensuring flexibility. This decision will affect the API results, which in turn impacts the user experience.

From the user's perspective, they probably care less about whether the `GetDocuments API` is consistent with the `GetDocument API` and more about whether the document displays correctly on their screen.

With this in mind, I agree with your opinion that `approach A` is appropriate. I will make the logic of both the in-memory and mongo implementations follow the same flow as `approach A` and add tests accordingly.